test/test_tag_helper.py

```diff
@@ -34,7 +34,7 @@ def test_add_tag(value, expected_tags, tags_were_added):
     tag = "blah"
     safelist = {"match": {"domain": ["blah.ca"]}}
     assert add_tag(res_sec, tag, value, safelist) == tags_were_added
-    assert res_sec.tags == expected_tags
+    assert res_sec._tag_sets == expected_tags
```

Comment:
Could you also validate that the expected_tags didn't change from an API point of view? Maybe rename that new one as internal_expected_tags, and keep the old one as it was?

Reply:
this test is no longer part of assemblyline-v4-service but I have switched to an implementation that preserves insertion order and added a test that verifies that insertion order is preserved.